extension_service/datastore/providers/firestore.py

```diff
@@ -0,0 +1,281 @@
+        amenities_docs = self.__client.collection("amenities").stream()
+        flights_docs = self.__client.collection("flights").stream()
+
+        airports = []
+
+        async for doc in airport_docs:
+            airport_dict = doc.to_dict()
+            airport_dict["id"] = doc.id
+            airports.append(models.Airport.model_validate(airport_dict))
+
+        amenities = []
+        async for doc in amenities_docs:
+            amenity_dict = doc.to_dict()
+            amenity_dict["id"] = doc.id
+            amenities.append(models.Amenity.model_validate(amenity_dict))
+
+        flights = []
+        async for doc in flights_docs:
+            flight_dict = doc.to_dict()
+            flight_dict["id"] = doc.id
+            flights.append(models.Flight.model_validate(flight_dict))
+
+        return airports, amenities, flights
+
+    async def get_airport_by_id(self, id: int) -> Optional[models.Airport]:
+        query = self.__client.collection("airports").where(
+            filter=FieldFilter("id", "==", id)
+        )
+        return models.Airport.model_validate(query.stream().to_dict())
+
+    async def get_airport_by_iata(self, iata: str) -> Optional[models.Airport]:
+        query = self.__client.collection("airports").where(
+            filter=FieldFilter("iata", "==", iata)
+        )
+        return models.Airport.model_validate(query.stream().to_dict())
+
+    async def search_airports(
+        self,
+        country: Optional[str] = None,
+        city: Optional[str] = None,
+        name: Optional[str] = None,
+    ) -> list[models.Airport]:
+        query = self.__client.collection("airports")
+
+        if country is not None:
+            query = query.where("country", "==", country)
+
+        if city is not None:
+            query = query.where("city", "==", city)
+
+        if name is not None:
+            query = query.where("name", ">=", name).where("name", "<=", name + "\uf8ff")
+
+        docs = query.stream()
+        airports = []
+        async for doc in docs:
+            airports.append(models.Airport.model_validate(dict(doc)))
+        return airports
+
+    async def get_amenity(self, id: int) -> Optional[models.Amenity]:
+        query = (
+            self.__client.collection("amenities")
+            .where(filter=FieldFilter("id", "==", id))
+            .select(
+                "id",
+                "name",
+                "description",
+                "location",
+                "terminal",
+                "category",
+                "hour",
+                "content",
+                "embedding",
+            )
+        )
+        return models.Amenity.model_validate(query.stream().to_dict())
+
+    async def amenities_search(
+        self, query_embedding: list[float], similarity_threshold: float, top_k: int
+    ) -> list[models.Amenity]:
+        query = (
+            self.__client.collection("amenities")
+            .where("embedding", ">", 1 - similarity_threshold)
```

Comment:
I don't follow this -- does Firestore support Vector search? I don't think this is doing what we want it to do. 

This might be one of the gaps where we need to find an alternative or not use until we can update with semantic similarity search.

Reply:
Sorry this was just a place holder and I missed it...Maybe we can try using https://extensions.dev/extensions/googlecloud/firestore-semantic-search? This extension seem like a plausible alternative?

Reply:
I'm not sure how well that will work -- it looks like it's creating the embedding as part of the query and using Vertex Vector Search on the backend... we would need to set up Vector Search which is a lot more expensive than just firestore. 

My gut is we will just need to throw not implemented and figure out how to demonstrate that in the front end.

Reply:
Raise Unimplemented Error for now.